Application/Telegram/UserStateStorage.cs

```diff
@@ -45,7 +50,37 @@ public static void UserStateUpdate(long telegramId, UserState newState)
         public static void AddUser(long telegramId, UserState state, UserRole role)
         {
 
-            userInfo.TryAdd(telegramId, new UserInfo() { CurrentState = state, Role = role });
+            userInfo.TryAdd(telegramId, new UserInfo() { CurrentState = state, Role = role});
+        }
+
+        public static void AddRecordToUserMessages(long telegramId, List<int> messageList)
+        {
+            userMessages.TryAdd(telegramId, messageList);
+        }
+        public static void AddMessage(long telegramId,int messageId)
+        {
+            var userMessageList = GetUserMessages(telegramId);
+            userMessageList.Add(messageId);
+            userMessages[telegramId] = userMessageList;
+        }
+
+        public static void RemoveMessages(long telegramId)
+        {
+            var userMessageList = GetUserMessages(telegramId);
```

Comment:
it's better to remove full list with TryDelete and do not keep empty list (if user will need, we will create new one)

Reply:
I will try to improve this later